### PR TITLE
Reset thread name in thread pool

### LINKED ISSUE #36114: Reset thread names in thread pool
**Describe the issue**
We observe many obsolete names of idle threads like `Formatter`.

**Solution**
Reset thread name after a job is done in thread pool.

### PR DESCRIPTION
**Pull request by alexey-milovidov**
### Changelog category (leave one):
- Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Reset thread name in `ThreadPool` to `ThreadPoolIdle` after job is done. This is to avoid displaying the old thread name for idle threads. This closes #36114.

### CODE REVIEW

**Comment by alexey-milovidov** [comment]
CC @azat

**Comment by alexey-milovidov** [comment]
Motivation: https://pastila.nl/?0523d2a9/70ceee4306afbdce91d4ab30ab86dfe6

**Comment by alexey-milovidov** [comment]
@mergifyio update

File: src/Common/ThreadPool.cpp [hunk 1]
```diff
@@ -243,6 +244,9 @@ void ThreadPoolImpl<Thread>::worker(typename std::list<Thread>::iterator thread_
 
     while (true)
     {
+        /// This is inside the loop to also reset previous thread names set inside the jobs.
+        setThreadName("ThreadPoolIdle");
```
[the remark this answers is not in the document]
Comment (azat) [reply src/Common/ThreadPool.cpp:248]: Maybe just set it to `argv[0][:15]` (i.e. `clickhouse-serv`, `clickhouse-clie`, ...) then?

Another option is to change thread name before executing something to i.e. `Wrk`, and after execution had been finished to i.e. `Idle`, but this looks too superfluous, first option sounds better.

[the remark this answers is not in the document]
Comment (tavplubix) [reply src/Common/ThreadPool.cpp:248]: > Maybe just set it to argv[0][:15] (i.e. clickhouse-serv, clickhouse-clie, ...) then?

Yep, or to just `ThreadPool`

[the remark this answers is not in the document]
Comment (alexey-milovidov) [reply src/Common/ThreadPool.cpp]: Ok.

**Review by tavplubix — APPROVED** [review]

**Comment by tavplubix** [comment]
ClickHouse special build check (actions) - some spot instances were killed, ok
Stateless tests (release, s3 storage, actions) - `01171_mv_select_insert_isolation_long` - `File segment can be completed only by downloader`, cc: @kssenii 
Stress test (memory, actions) - still don't have stacktraces, maybe #36194 will help

**Pull request merged**